### ISSUE TITLE
nil passed as error parameter in many calls

### ISSUE THREAD

**Issue by RubioV** [mapbox/DEPRECATED-mapbox-ios-sdk#287]
Throughout the SDK, method calls with an `NSError` parameter are often called with `nil` instead of `NULL`. Example:

`id metadata = [NSJSONSerialization JSONObjectWithData:metadataData options:0 error:nil];`

According to the Apple documentation, the proper way to call such methods is with `NULL` if you're not interested in the error:

_Always test to see if the parameter is non-NULL before assigning an object to it. And never assign nil to an error parameter._

I'll submit a pull request in the next few days to correct this.

**Comment by incanus** [comment]
Hmm, I had read someplace (IIRC) that `nil` was ok for this sort of thing due to compiler settings. I'll see if I can dig that up.

**Comment by RubioV** [comment]
Yes, in fact, after I created this post, I tested it and as far as I can tell there is no difference. So passing `nil` to a method and then comparing it to `NULL` will render true. I remembered the statement in the Apple docs that explicitly tell you not to mix them. Anyway, I think it doesn't hurt to fix it, just to be in line with Apple's recommendations. 

A little more info in this stackoverflow post:
http://stackoverflow.com/questions/2461617/are-null-and-nil-equivalent
